src/runtime-rs/crates/resource/src/share_fs/share_virtio_fs_standalone.rs

```diff
@@ -0,0 +1,200 @@
+        let source_path = get_host_ro_shared_path(&self.config.id);
+        if !source_path.exists() {
+            return Err(anyhow!("The virtiofs shared path didn't exist"));
+        }
+
+        let mut args: Vec<String> = vec![
+            String::from("-f"),
+            String::from("-o"),
+            format!("vhost_user_socket={}", sock_path),
+            String::from("-o"),
+            format!("source={}", source_path.to_str().unwrap()),
+            String::from("-o"),
+            format!("cache={}", self.config.virtio_fs_cache),
+        ];
+
+        if !self.config.virtio_fs_extra_args.is_empty() {
+            let mut extra_args: Vec<String> = self
+                .config
+                .virtio_fs_extra_args
+                .iter()
+                .map(String::from)
+                .collect();
+            args.append(&mut extra_args);
+        }
+
+        Ok(args)
+    }
+
+    async fn setup_virtiofsd(&self) -> Result<()> {
+        let sock_path = generate_sock_path(&self.config.jail_root);
+        let args = self.virtiofsd_args(&sock_path).context("virtiofsd args")?;
+
+        let mut cmd = Command::new(&self.config.virtio_fs_daemon);
+        let child_cmd = cmd.args(&args).stderr(Stdio::piped());
+        let child = child_cmd.spawn().context("spawn virtiofsd")?;
+
+        // update virtiofsd pid{
+        {
+            let mut inner = self.inner.write().await;
+            inner.pid = child.id();
+        }
+
+        let (tx, mut rx): (Sender<Result<()>>, Receiver<Result<()>>) = channel(100);
+        tokio::spawn(run_virtiofsd(child, tx));
+
+        // TODO: support timeout
+        match rx.recv().await.unwrap() {
+            Ok(_) => {
+                info!(sl!(), "start virtiofsd successfully");
+                Ok(())
+            }
+            Err(e) => {
+                error!(sl!(), "failed to start virtiofsd {}", e);
+                self.shutdown_virtiofsd()
+                    .await
+                    .context("shutdown_virtiofsd")?;
+                Err(anyhow!("failed to start virtiofsd"))
+            }
+        }
+    }
+
+    async fn shutdown_virtiofsd(&self) -> Result<()> {
+        let mut inner = self.inner.write().await;
+
+        if let Some(pid) = inner.pid {
+            let pid = ::nix::unistd::Pid::from_raw(pid as i32);
+            if let Err(err) = ::nix::sys::signal::kill(pid, nix::sys::signal::SIGKILL) {
+                if err != ::nix::Error::Sys(nix::errno::Errno::ESRCH) {
+                    return Err(anyhow!("failed to kill virtiofsd pid {} {}", pid, err));
+                }
+            }
+        }
+        inner.pid = None;
+
+        Ok(())
+    }
+}
+
+async fn run_virtiofsd(mut child: Child, tx: Sender<Result<()>>) -> Result<()> {
+    let stderr = child.stderr.as_mut().unwrap();
+    let mut stderr_reader = BufReader::new(stderr);
+
+    loop {
+        let mut buffer: String = String::new();
+        match stderr_reader.read_line(&mut buffer).await {
```

Comment:
What about using for + lines() to simplify.

Reply:
Thanks for your review, for `tokio::io` read I'am not find `for + lines()`, change to `while + lines()`.